packages/hash/api/src/model/block.model.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Entity,
+  EntityConstructorArgs,
+  Block,
+  User,
+  EntityExternalResolvers,
+  UnresolvedGQLEntityType,
+} from ".";
+import { DBClient } from "../db";
+import { DbBlockProperties, EntityType } from "../db/adapter";
+import { SystemTypeName, Block as GQLBlock } from "../graphql/apiTypes.gen";
+
+export type BlockExternalResolvers = EntityExternalResolvers | "entity"; // entity resolved in `src/graphql/resolvers/block/linkedEntities.ts`
+
+export type UnresolvedGQLBlock = Omit<GQLBlock, BlockExternalResolvers> & {
+  entityType: UnresolvedGQLEntityType;
+};
+
+export type BlockConstructorArgs = {
+  properties: DbBlockProperties;
+  outgoingEntityIds?: string[];
+} & EntityConstructorArgs;
+
+class __Block extends Entity {
+  properties: DbBlockProperties;
+
+  parentEntityId?: string;
+
+  constructor(args: BlockConstructorArgs) {
+    super(args);
+    this.properties = args.properties;
+    this.parentEntityId = args.outgoingEntityIds?.[0] ?? undefined;
+  }
+
+  static async getEntityType(client: DBClient): Promise<EntityType> {
+    const blockEntityType = await client.getSystemTypeLatestVersion({
+      systemTypeName: "Block",
+    });
+    return blockEntityType;
+  }
+
+  static async createBlock(
+    client: DBClient,
+    params: {
+      accountId?: string;
+      properties: DbBlockProperties;
+      createdBy: User;
+      blockEntity: Entity;
+    },
+  ): Promise<Block> {
+    const { blockEntity, properties, createdBy, accountId } = params;
+
+    const entity = await Entity.createEntityWithLinks(client, {
+      user: createdBy,
+      accountId: accountId ?? createdBy.accountId,
+      entityDefinition: {
+        entityProperties: properties,
+        entityType: {
+          systemTypeName: SystemTypeName.Block,
+        },
+        linkedEntities: [
+          {
+            path: "$.entity",
```

Comment:
@CiaranMn @nathggns @thehabbos007 I've called this link `entity` to be consistent with how this link was previously stored in the properties of a block, but think this name could lead to confusion.

The only other ideas I had were:
```
(Block)-[blockEntity]->(Entity)
(Block)-[props]->(Entity) // as this entity is what we're passing to the block? Kind of like react props?
```
Any suggestions or thoughts on this?

Reply:
The linked `Entity` would be considered the entity that contains the data attached to the block, right? In that case `contentEntity` or `dataEntity` might make sense, but would be a new "term". I agree that simply `entity` could lead to confusion.

Reply:
So not that - potentially any of the `[prefix]Entity` mentioned.

Reply:
In an internal discussion `data` was agreed upon:
```
(Block)-[data]->(Entity)
```
Have addressed this in this commit: https://github.com/hashintel/hash/pull/303/commits